### ISSUE TITLE
Driving skill accumulates experience needlessly slowly

### ISSUE THREAD

**Issue by graysage1** [CleverRaven/Cataclysm-DDA#12200]
I have 'Fast learner' and 'Stylish' traits.  I'm wearing lots of stylish gear so my focus is routinely over 100.  Given all these favorable conditions, and given that I've been driving my motorized wheelbarrow for close to 30 in-game days, it seems silly that my driving skill hasn't hit 2 yet.  This to me seems absurd.

It's crazy that you can learn driving faster by reading a book than by actual driving.  It makes little sense to me.

**Comment by ejseto** [comment]
Like most things in life, we learn more from failure than success. Simply driving around has a 1-in-10 chance per turn to give ONE driving xp. Failure (skidding), on the other hand, gives speed/5 driving xp when you regain control.

**Comment by Coolthulhu** [comment]
Driving is probably the biggest offender when it comes to unfinished skills.
It's usefulness currently pretty much caps at 4 (no fumbling with controls), the skill gain doesn't depend on speed and is abysmally low.

**Comment by graysage1** [comment]
@Coolthulhu I agree, so can we raise it please?  It's not like when I get driving skill 10 I will suddenly be overpowered. I'll just not skid anymore.  Big deal.  It's just a creature comfort, so why deny it so hard?

**Comment by graysage1** [comment]
>Failure (skidding), on the other hand, gives speed/5 driving xp when you regain control.

If true I really don't like this kind of sneaky undocumented behavior, personally.  It's also not universally true that we learn more from failure than success in every single case.  For example, if I try to pick up 1000 lbs barbell, and it results in a failure, I don't learn more than doing a proper *successful* workout with a 100lbs barbell.

**Comment by ejseto** [comment]
You're not learning anything from picking up a barbell. I know "muscle memory" is a thing, but muscles do not literally learn, that implies intelligence. And I didn't say it's universally true, I said "like most things."

**Comment by graysage1** [comment]
>You're not learning anything from picking up a barbell.

Yes I am.

**Comment by graysage1** [comment]
I was thinking about this yesterday when I realized that even if we go with the "you learn more from failures" quip as a guide, then shooting while driving should earn major XP for driving in addition to shooting.  And as it happened, in my testing I've been doing lots of drive-by's, and my XP gain was still extremely slow.

I don't think it's a good idea to encourage players to skid.  Encouraging suboptimal play to maximize XP doesn't seem like a good game design for such a game.  Instead XP should accumulate optimally from ordinary or skilled play, but not from weird play, like forcing yourself to skid deliberate.

**Comment by Shoes01** [comment]
Perhaps fumbling should give some exp, based on speed. Driving + shooting should increase fumbling and thus improve driving.

**Comment by Coolthulhu** [comment]
Make fumbling chance scale with speed, make fumbling give xp.
Later on add handling difficulty, so that huge trucks need more skill at the same speed than motorbikes.

**Comment by Zireael07** [comment]
I like the way you're thinking here, @Shoes01 and @Coolthulhu.

**Comment by ejseto** [comment]
At the risk of derailing and responding to a potential troll, please tell me what you *learn* by picking up a barbell. I am truly curious what exactly you *learn* from manual labor that has no purpose other than exercise.

**Comment by DeNarr** [comment]
I think hitting zombies or running over things should give a good boost to driving skill, and it would be really nice if high skill in driving would help mitigate the damage you do to your car when performing those activities. Knowing how to hit things in the best spot should help your car live longer.

**Comment by i2amroy** [comment]
Really this just needs a quick fix of making players gain a single point of driving skill from driving straight (which doesn't currently make you gain anything).

**Comment by ejseto** [comment]
Don't you gain from straight driving? I didn't look that closely but under the pause() function (which would constitute driving straight) you have a 10% chance to gain ONE xp. Granted that's close to nothing, but it's still something.

**Comment by Zireael07** [comment]
AFAIK no, only turning increases driving xp.

**Comment by ejseto** [comment]
Under player::pause() if you are in control of the vehicle and your velocity is not zero, you have a 1-in-10 chance to gain 1 xp.

**Comment by Zireael07** [comment]
LOL, 1-in-10 to gain 1 xp? No wonder it's negligible and completely unnoticeable in practice. Get rid of the 1-in-10 chance and increase that xp gain. I'm thinking 25% of what is gained by turning....

**Comment by Coolthulhu** [comment]
The gain for turning should be removed. It promotes "wiggling", which is tedious and also silly.

**Comment by graysage1** [comment]
> It promotes "wiggling", which is tedious and also silly.

I agree.

**Comment by DavidKeaton** [comment]
mass of vehicle could come into play. used to work at a car place, and had to drive both small and p large vehicles. I can say that a heavier vehicle def taught me to drive both those and smaller vehicles better, as my knowledge of how different size vehicles reacted to acceleration and decelerating behaved, as well as what speeds were safe to turn at what angle.

**Comment by ejseto** [comment]
Well it wasn't quite so insignificant until someone decided required xp to level up needed to be squared.

**Comment by Coolthulhu** [comment]
It was. Getting 10 mechanics and 10 in melee was easier than getting 4 in driving.

**Comment by ejseto** [comment]
I didn't say it wasn't. I said it wasn't AS insignificant. Now it's even more so.
